Sage/carbon
# chore: convert remaining javascript utils to typescript

### Proposed behaviour

- Convert remaining internal util files from javascript to typescript
- Refactor directory structure for ease of finding utils

### Checklist




### Additional context

`src/__spec_helper__/enzyme-snapshot-helper.js` has not been updated, given plans on our roadmap to move away from enzyme to React Testing Library

### Testing instructions

- Check `npm run type-check` passes without errors
- Ensure tests all pass - particularly for files which have had import paths updated
